src/main/java/com/bookappstore/repository/book/spec/BookSpecificationBuilder.java

```diff
@@ -69,6 +69,12 @@ public Specification<Book> build(BookSearchParametersDto bookSearchParametersDto
                     .getSpecification(bookSearchParametersDto.description()));
         }
 
+        if (bookSearchParametersDto.categories() != null && bookSearchParametersDto.categories()
+                .length > 0) {
+            specification = specification.and(bookSpecificationProviderManager
+                    .getSpecificationProvider("category")
+                    .getSpecification(bookSearchParametersDto.categories()));
+        }
```

Comment:
I dont like how your  build method works. There's too many duplicates. Think of some abstraction

Reply:
Thanks, good tip